modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -0,0 +1,281 @@
+/obj/structure/reagent_crafting_bench
+	name = "forging work bench"
+	desc = "A crafting bench used to create forging items."
+	icon = 'modular_skyrat/modules/reagent_forging/icons/obj/forge_structures.dmi'
+	icon_state = "crafting_bench_empty"
+
+	anchored = TRUE
+	density = TRUE
+
+	///the current goal item that is obtainable
+	var/goal_item_path
+	///the name of the goal item
+	var/goal_name
+	///the amount of chains within the bench
+	var/chain_amount = 0
+	///the amount of chains required
+	var/required_chain = 0
+	///the amount of plates within the bench
+	var/plate_amount = 0
+	///the amount of plates required
+	var/required_plate = 0
+	///the amount of coils within the bench
+	var/coil_amount = 0
+	///the amount of coils required
+	var/required_coil = 0
+	///the amount of hits required to complete the item
+	var/required_hits = 0
+	///the current amount of hits
+	var/current_hits = 0
+	///the amount of wood currently stored
+	var/current_wood = 0
+	//so we can't just keep being hit without cooldown
+	COOLDOWN_DECLARE(hit_cooldown)
+	///the choices allowed in crafting
+	var/list/allowed_choices = list(
+		"Chain Helmet",
+		"Chain Armor",
+		"Chain Gloves",
+		"Chain Boots",
+		"Plated Boots",
+		"Horseshoes",
+		"Ring",
+		"Collar",
+		"Handcuffs",
+		"Borer Cage",
+		"Pavise Shield",
+		"Buckler Shield",
+		"Coil",
+	)
+	///the paths of the allowed choices
+	var/static/list/choice_paths = list(
```

Comment:
Could this potentially be an assoc list, allowing you to combine it with `allowed_choices` above?